src/components/StatedGraph.js

```diff
@@ -1,14 +1,19 @@
 import React from 'react';
 import { Line } from 'react-chartjs-2';
+import { observer } from 'mobx-react-lite';
 import moment from 'moment';
 
-const Chart = (values) => {
-  let currentMonthDates = Array.from(
+
+const StatedGraph = observer(({ balanceService }) => {
```

Comment:
This class has now two responsibilities.
One is extracting data from mobX and handling it.
Second is using data to generate chart.

I'd recommend it to divide it into StatedGraph and ComplexGraph.

StatedGraph:
- should be observable
- should extract data from mobx (aka using getBalances())
- pass data to ComplexGraph (taking data from ^ and passing to Graph)

ComplexGraph:
- should input some data (it shouldnt care where it comes from, it is just pure React component)
- should generate Graph using this data

This approach gives us:
- If we wanted to move to redux in the future, we could reuse ComplexGraph
- We separate mental load (in StatedGraph we only think about getting the data and passing it, in ComplexGraph we only think about displaying the data)

Reply:
Btw. Data processing should probably be left in ComplexGraph, because there in the future we may change the lib used for displaying graph and we would have to change the data transformations for it. So it is safer to keep it close.